### PR TITLE
Review crud + UserReviews + Search book/author

### LINKED ISSUE #48: [backend] Create endpoint to search book/author
Ac:
- endpoint for book/author is added
- endpoint parameters: enum: book/author, string: text
- service should search for the exact entity (search by text)

Assumption:
N/A

### LINKED ISSUE #71: [B] Create CRUD for reviews
Should be able to query by user's ID

### LINKED ISSUE #103: [B] Create endpoint that returns user's reviews

### PR DESCRIPTION
**Pull request by kris875**
closes #71 
closes #103 
closes #48

### CODE REVIEW

File: server/src/main/java/com/uj/projects/booksplatform/review/controller/ReviewController.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    @GetMapping(params = "user")
+    public List<ReviewDto> getReviewsByUser(@RequestParam(name = "user") Integer userId){
+        return reviewService.getByUser(userId).stream().map(reviewMapper::reviewToReviewDto).collect(Collectors.toList());
+    }
+
+    @PostMapping
+    public ReviewDto createReview(@RequestBody ReviewDto reviewDto){
+        reviewDto.setId(0);
```
Comment (SzymonSmykala) [inline server/src/main/java/com/uj/projects/booksplatform/review/controller/ReviewController.java:49]: do we need to set it? If not, please remove

File: server/src/main/java/com/uj/projects/booksplatform/review/mapper/ReviewMapper.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+@Mapper(componentModel = "spring", injectionStrategy = InjectionStrategy.CONSTRUCTOR)
+public interface ReviewMapper {
+    @Mapping(source = "book.id", target = "book")
+    @Mapping(source = "user.id", target = "user")
+    ReviewDto reviewToReviewDto(Review review);
+
+    @Mapping(source = "book", target = "book.id")
+    @Mapping(source = "user", target = "user.id")
+    Review reviewDtoToReview(ReviewDto reviewDto);
+
```
Comment (SzymonSmykala) [inline server/src/main/java/com/uj/projects/booksplatform/review/mapper/ReviewMapper.java:18]: remove empty redundant empty lines

File: server/src/main/java/com/uj/projects/booksplatform/review/repository/ReviewRepository.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import com.uj.projects.booksplatform.review.entity.Review;
+import com.uj.projects.booksplatform.user.entity.User;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+@Repository
+public interface ReviewRepository extends JpaRepository<Review, Integer> {
+    List<Review> findReviewsByBook(Book book);
```
Comment (SzymonSmykala) [inline server/src/main/java/com/uj/projects/booksplatform/review/repository/ReviewRepository.java:13]: please rename to findReviewsBy(...) both methods

File: server/src/main/java/com/uj/projects/booksplatform/review/repository/ReviewRepository.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    List<Review> findReviewsByUser(User user);
+    Integer countByBook(Book book);
```
Comment (SzymonSmykala) [inline server/src/main/java/com/uj/projects/booksplatform/review/repository/ReviewRepository.java:15]: countBy(...)

File: server/src/main/java/com/uj/projects/booksplatform/review/service/ReviewService.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+import java.util.List;
+
+public interface ReviewService {
+    List<Review> getAll();
+    Review create(Review review);
+    void delete(Integer id);
+    Review update(Review review);
+    Review getById(Integer id);
+    List<Review> getByBook(Integer id);
```
Comment (SzymonSmykala) [inline server/src/main/java/com/uj/projects/booksplatform/review/service/ReviewService.java:13]: getByBookID 
getByUserID
getNumberOfReviewsByBookID

File: server/src/main/java/com/uj/projects/booksplatform/review/repository/ReviewRepository.java (hunk 3 above)
Comment (MidoriyaAmongProgrammers) [reply server/src/main/java/com/uj/projects/booksplatform/review/repository/ReviewRepository.java:13]: I can't do this. These are the default names from JPA

Comment (kris875) [reply server/src/main/java/com/uj/projects/booksplatform/review/repository/ReviewRepository.java:13]: I can't do this. These are the default names from JPA

Comment (SzymonSmykala) [reply server/src/main/java/com/uj/projects/booksplatform/review/repository/ReviewRepository.java:13]: wow

File: server/src/main/java/com/uj/projects/booksplatform/review/repository/ReviewRepository.java (hunk 4 above)
Comment (kris875) [reply server/src/main/java/com/uj/projects/booksplatform/review/repository/ReviewRepository.java:15]: I can't do this. These are the default names from JPA

**Pull request merged**